<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="description" content="Research in DNA Read Mapping done with the SAFARI group at Carnegie Mellon University.">
        <meta name="author" content="Sunny Nahar">
        <title>Research in DNA Read Mapping</title>
        <link rel="canonical" href="https://snnynhr.github.io/dna-sequencing">
        <link rel="stylesheet" href="css/bootstrap.min.css" type="text/css">
        <link href="https://fonts.googleapis.com/css?family=Montserrat:400,300,600,700,800" rel="stylesheet">
        <link rel="stylesheet" href="font-awesome/css/font-awesome.min.css" type="text/css">
        <link rel="stylesheet" href="css/animate.critical.css" type="text/css">
        <link rel="stylesheet" href="css/creative.css" type="text/css">
    </head>
    <body id="page-top">
        <nav id="mainNav" class="navbar navbar-default navbar-fixed-top">
            <div class="container-fluid">
                <div class="navbar-header">
                    <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                    <span class="sr-only">Toggle navigation</span>
                    <i class="fa fa-2x fa-bars"></i>
                    </button>
                    <a class="navbar-brand page-scroll" href="#page-top">DNA Read Mapping</a>
                </div>
                <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                    <ul class="nav navbar-nav navbar-right">
                        <li><a class="page-scroll" href="#about">About</a></li>
                        <li><a class="page-scroll" href="#researchf16">F16</a></li>
                        <li><a class="page-scroll" href="#researchs16">S16</a></li>
                        <li><a class="page-scroll" href="#researchf15">F15</a></li>
                        <li><a class="page-scroll" href="#researchs15">S15</a></li>
                        <li><a class="page-scroll" href="#contact">Contact</a></li>
                    </ul>
                </div>
            </div>
        </nav>
        <header>
            <div class="header-content">
                <div class="header-content-inner">
                    <h1>Accelerating DNA Read Mapping</h1>
                    <hr>
                    <p>Improving algorithms for DNA read mapping to produce fast and efficient DNA mappers.</p>
                    <a href="#about" class="btn btn-primary btn-xl page-scroll">
                        <i class="fa fa-2x fa-arrow-down animated"></i>
                    </a>
                </div>
            </div>
        </header>
        <section class="bg-dark" id="about">
            <div class="container">
                <div class="row">
                    <div class="col-lg-8 col-lg-offset-2 text-center">
                        <h2 class="section-heading">What is DNA Sequencing?</h2>
                        <hr class="light">
                    </div>
                    <div class="col-lg-8 col-lg-offset-2">
                        <p class="text-faded">DNA Sequencing is the process of determining the sequence of base-pairs composing DNA. Shotgun sequencing splits DNA into small chunks called reads, and reads need to be reassembled or mapped into the overall DNA. Mapper speed remains a bottleneck in DNA read mapping. Processing and aligning billions of reads take a substantial amount of time, and there is a quest to build faster and better algorithms to improve runtime and throughput. In order to further the use DNA sequencing technology, it’s important to make it accessible and efficient.</p>
                    </div>
                </div>
            </div>
        </section>
        <section id="researchf16" class="bg-primary">
            <div class="container">
                <div class="row">
                    <div class="col-lg-12 text-center">
                        <h2 class="section-heading">Fall 2016</h2>
                        <hr class="light">
                    </div>
                </div>
                <div class="row">
                    <h3 class="section-subheading"> <b>LEAP: Extending Landau-Vishkin</b></h3>
                    <p class="text-faded">
                        Approximate String Matching is a pivotal problem in the field of computer science. It serves as an integral component for many string algorithms, most notably, DNA read mapping and alignment. The improved LV algorithm proposes an improved dynamic programming strategy over the banded Smith-Waterman algorithm but suffers from support of a limited selection of scoring schemes. In this paper, we propose the Leaping Toad problem, a generalization of the approximate string matching problem, as well as LEAP, a generalization of the Landau-Vishkin’s algorithm that solves the Leaping Toad problem under a broader selection of scoring schemes.
                    </p>
                </div>
                <div class="container text-center">
                    <div class="row">
                        <div class="col-lg-12 col-md-8 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-file-text wow bounceIn text-default"></i>
                                <h3></h3>
                                <a href="https://www.biorxiv.org/content/biorxiv/early/2017/05/02/133157.full.pdf" class="btn btn-default btn-xl wow">Research Paper</a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
        <section id="researchs16">
            <div class="container">
                <div class="row">
                    <div class="col-lg-12 text-center">
                        <h2 class="section-heading">Spring 2016</h2>
                        <hr class="primary">
                    </div>
                </div>
                <div class="row">
                    <h3 class="section-subheading"><b>Parallelizing DNA Mapping Architecture</b></h3>
                    <p>
                        Single processor compute capabilities have not improved significantly, and with the advent
                        of multi-core and multi-threaded machines, parallelism is important in achieving the next
                        generation of performance. Many aspects of DNA read mapping are inherently parallel, so
                        this makes it a viable approach for achieving improvements in runtime.
                        This research also focuses on optimizing and parallelizing a component of DNA read
                        mapping pipeline, seed selection. We want to determine how efficiently we can distribute
                        work over cores to achieve a large speedup.
                    </p>
                </div>
                <div class="container text-center">
                    <div class="row">
                        <div class="col-lg-4 col-md-5 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-file-text wow bounceIn text-primary"></i>
                                <h3></h3>
                                <a href="cmu/spring16/springfinalreport.pdf" class="btn btn-primary btn-xl wow">Report</strong></a>
                            </div>
                        </div>
                        <div class="col-lg-4 col-md-5 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-line-chart wow bounceIn text-primary"></i>
                                <h3></h3>
                                <a href="cmu/spring16/presentation.pdf" class="btn btn-primary btn-xl wow">Presentation</strong></a>
                            </div>
                        </div>
                        <div class="col-lg-4 col-md-5 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-image wow bounceIn text-primary"></i>
                                <h3></h3>
                                <a href="cmu/spring16/poster.pdf" class="btn btn-primary btn-xl wow">Poster</strong></a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
        <section id="researchf15" class="bg-primary">
            <div class="container">
                <div class="row">
                    <div class="col-lg-12 text-center">
                        <h2 class="section-heading">Fall 2015</h2>
                        <hr class="light">
                    </div>
                </div>
                <div class="row">
                    <h3 class="section-subheading"> <b>Improving Seed Selection in DNA Read Mapping</b></h3>
                    <p class="text-faded">
                        In this research, we focus on improving the complexity and runtime of seed-and-extend based mappers. Seed-and-extend mappers align reads by splitting reads into smaller seeds. These seeds are searched in the reference and used to align the reads. The complexity of the alignment is in correlation with the how often the seeds occur in the reference, or the frequency of the seeds. If seeds occur more frequently, a larger number of locations need to checked for potential mapping sites of the read, which increases complexity. Therefore it is imperative that we choose seeds which have the lowest sum of frequencies to minimize the number of mapping locations, and hence computation.
                    </p>
                    <p class="text-faded">
                        Therefore the goal is to develop state-of-the-art seed selection schemes which are able to select seeds from a read in a memory efficient and cache efficient manner while minimizing the sum of the frequency of the seeds. These new heuristics and algorithms will be compared with existing seed selection schemes found in seed-and-extend mappers such as Hobbes and
                        FastHash, among others.
                    </p>
                </div>
                <div class="container text-center">
                    <div class="row">
                        <div class="col-lg-6 col-md-8 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-file-text wow bounceIn text-default"></i>
                                <h3></h3>
                                <a href="cmu/fall15/AcceleratingDNAReadMapping.pdf" class="btn btn-default btn-xl wow">Report</a>
                            </div>
                        </div>
                        <div class="col-lg-6 col-md-8 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-image wow bounceIn text-default"></i>
                                <h3></h3>
                                <a href="cmu/fall15/poster.pdf" class="btn btn-default btn-xl wow">Poster</a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
        <section id="researchs15" class="bg-light">
            <div class="container">
                <div class="row">
                    <div class="col-lg-12 text-center">
                        <h2 class="section-heading">Spring 2015</h2>
                        <hr class="primary">
                    </div>
                </div>
                <div class="row">
                    <h3 class="section-subheading"><b>Optimal Seeds Solver</b></h3>
                    <p>
                        Optimizing seed selection is an important problem in read mapping. The number of non-overlapping seeds a mapper selects determines the sensitivity of the mapper while the total frequency of all selected seeds determines the speed of the mapper. Modern seed-and-extend mappers usually select seeds with either an equal and fixed-length scheme or with an inflexible placement scheme, both of which limit the ability of the mapper in selecting less frequent seeds to speed up the mapping process. Therefore, it is crucial to develop a new algorithm that can adjust both the individual seed length and the seed placement, as well as derive less frequent seeds.
                    </p>
                </div>
                <div class="row">
                    <h3 class="section-subheading"><b>Paper</b></h3>
                    <p> We published a paper in the <a href="https://bioinformatics.oxfordjournals.org/">Oxford Bioinformatics Journal</a>. </p>
                </div>
                <div class="container text-center">
                    <div class="row">
                        <div class="col-lg-4 col-md-5 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-github wow bounceIn text-primary"></i>
                                <h3></h3>
                                <a href="https://github.com/CMU-SAFARI/optimal-seed-solver" class="btn btn-primary btn-xl wow">View On <strong>GitHub</strong></a>
                            </div>
                        </div>
                        <div class="col-lg-4 col-md-5 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-file-text wow bounceIn text-primary"></i>
                                <h3></h3>
                                <a href="https://bioinformatics.oxfordjournals.org/content/early/2016/01/20/bioinformatics.btv670.abstract" class="btn btn-primary btn-xl wow">Journal Link</strong></a>
                            </div>
                        </div>
                        <div class="col-lg-4 col-md-5 text-center">
                            <div class="service-box">
                                <i class="fa fa-4x fa-file-text wow bounceIn text-primary"></i>
                                <h3></h3>
                                <a href="https://arxiv.org/pdf/1506.08235v1.pdf" class="btn btn-primary btn-xl wow">Research Paper</strong></a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
        <section id="researchf14" class="bg-primary">
            <div class="container">
                <div class="row">
                    <div class="col-lg-12 text-center">
                        <h2 class="section-heading">Fall 2014</h2>
                        <hr class="light">
                    </div>
                </div>
                <div class="row">
                    <p class="text-faded"> Began work in DNA Read Mapping.</p>
                </div>
            </div>
        </section>
        <section id="contact">
            <div class="container">
                <div class="row">
                    <div class="col-lg-8 col-lg-offset-2 text-center">
                        <h2 class="section-heading">Authors</h2>
                        <hr class="primary">
                        <p>This work was done by <a class="emgen"> Sunny Nahar </a> under Prof. Onur Mutlu and Hongyi Xin as part of the <a href="https://www.ece.cmu.edu/~safari/">CMU Safari Group</a>.</p>
                    </div>
                </div>
            </div>
        </section>

        <!-- jQuery -->
        <script src="js/jquery.js"></script>
        <script src="js/bootstrap.min.js"></script>
        <script src="js/jquery.easing.min.js"></script>
        <script src="js/wow.min.js"></script>
        <!-- Custom Theme JavaScript -->
        <script src="js/creative.js"></script>
    </body>
</html>